libs/core/langchain_core/retrievers.py

```diff
@@ -262,6 +263,7 @@ async def _aget_relevant_documents(
             run_manager=run_manager.get_sync(),
         )
 
+    @deprecated("0.1.43", alternative="invoke", removal="0.2.0")
```

Comment:
what should go here?

Reply:
Best way to check is just to run it and see the warning gets printed. I tend to add this but otherwise lgtm
```suggestion
    @deprecated(since="0.1.43", alternative="invoke", removal="0.2.0")
```

Reply:
I think you also need to wrap the `self.get_relevant_documents()` call in `invoke` and `ainvoke` with some kind of warning suppressor (until we swap logic to be implemented in invoke and remove get_relevant_documents in 0.2)

Reply:
I'm not getting warnings on .invoke and I think it's because of the check for `is_caller_internal` in the decorator, but happy to add the warning suppressor if I'm mistaken (or if it's just prudent)

https://github.com/langchain-ai/langchain/blob/c909ae01523bdbfb32b5266d0de617643a82f355/libs/core/langchain_core/_api/deprecation.py#L145